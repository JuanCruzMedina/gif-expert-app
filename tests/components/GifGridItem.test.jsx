import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe("Test on component <GifGridItem>", () => {
  const title = "One punch battle";
  const url = "https://onepuch-gif.com/battle.png";
  test("Snapshot match", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test('should display the image with the indicated properties', () => { 
    render(<GifGridItem title={title} url={url}/>);
    const {src, alt} = screen.getByRole('img');
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });
  test('should display the title in the component', () => { 
    render(<GifGridItem title={title} url={url}/>);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
