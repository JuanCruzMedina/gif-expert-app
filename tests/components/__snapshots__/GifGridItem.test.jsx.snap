// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on component <GifGridItem> Snapshot match 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One punch battle"
      src="https://onepuch-gif.com/battle.png"
    />
    <p>
      One punch battle
    </p>
  </div>
</div>
`;
